doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst

```diff
@@ -0,0 +1,883 @@
+
+    // Set task properties
+    task.setProperty("group", arm_group_name);
+    task.setProperty("eef", eef_name);
+    task.setProperty("hand", hand_group_name);
+    task.setProperty("hand_grasping_frame", hand_frame);
+    task.setProperty("ik_frame", hand_frame);
+
+Now, we add an example stage to the node. The first line sets ``current_state_ptr`` to ``nullptr``; this creates a pointer to a stage such that we can re-use stage information in specific scenarios. This line is not used at this moment, but will be used later when more stages are added to the task. Next, we make a ``current_state`` stage (a generator stage) and add it to our task - this starts the robot off in its current state. Now that we've created the ``CurrentState`` stage, we save a pointer to it in the ``current_state_ptr`` for later use.
+
+.. code-block:: c++
+
+    moveit::task_constructor::Stage* current_state_ptr = nullptr;  // Forward current_state on to grasp pose generator
+    auto stage_state_current = std::make_unique<moveit::task_constructor::stages::CurrentState>("current");
+    current_state_ptr = stage_state_current.get();
+    task.add(std::move(stage_state_current));
+
+In order to plan any robot motions, we need to specify a solver. MoveIt Task Constructor has three options for solvers:
+
+ * ``PipelinePlanner`` uses the default MoveIt planner, which is typically OMPL.
+ * ``CartesianPath`` is used to move the end effector in a straight line in Cartesian space.
+ * ``JointInterpolation`` is a simple planner that interpolates between the start and goal joint states. It is typically used for simple motions as it is fast to plan, but may fail for complex motions.
+
+We also set some properties specific for to the Cartesian planner.
+
+.. code-block:: c++
+
+    auto sampling_planner = std::make_shared<moveit::task_constructor::solvers::PipelinePlanner>(node_);
+    auto interpolation_planner = std::make_shared<moveit::task_constructor::solvers::JointInterpolationPlanner>();
+
+    auto cartesian_planner = std::make_shared<moveit::task_constructor::solvers::CartesianPath>();
+    cartesian_planner->setMaxVelocityScaling(1.0);
+    cartesian_planner->setMaxAccelerationScaling(1.0);
+    cartesian_planner->setStepSize(.001);
+
+Now that we added in the planners, we can add a stage that will move the robot. The following lines use a ``MoveTo`` stage (a propagator stage). Since opening the hand is a relatively simple movement, we can use the joint interpolation planner. This stage plans a move to the "open hand" pose, which is a named pose defined in the :moveit_resources_codedir:`SRDF<panda_moveit_config/config/panda.srdf>` for the panda robot.
+
+.. code-block:: c++
+
+    auto stage_open_hand = std::make_unique<moveit::task_constructor::stages::MoveTo>("open hand", interpolation_planner);
+    stage_open_hand->setGroup(hand_group_name);
+    stage_open_hand->setGoal("open");
+    task.add(std::move(stage_open_hand));
+
+    return task;
+    }
```

Comment:
```suggestion
```

Reply:
The indentation was wrong, which makes this look pretty strange (hopefully fixed now), but I think that brace is needed (matches with line 410)